libtransmission/blocklist.cc

```diff
@@ -71,133 +54,109 @@ static void blocklistLoad(tr_blocklistFile* b)
     }
 
     tr_error* error = nullptr;
-    auto const fd = tr_sys_file_open(b->filename, TR_SYS_FILE_READ, 0, &error);
+    auto const fd = tr_sys_file_open(getFilename(), TR_SYS_FILE_READ, 0, &error);
     if (fd == TR_BAD_SYS_FILE)
     {
         tr_logAddWarn(fmt::format(
             _("Couldn't read '{path}': {error} ({error_code})"),
-            fmt::arg("path", b->filename),
+            fmt::arg("path", getFilename()),
             fmt::arg("error", error->message),
             fmt::arg("error_code", error->code)));
         tr_error_free(error);
         return;
     }
 
-    b->rules = static_cast<struct tr_ipv4_range*>(tr_sys_file_map_for_reading(fd, 0, byteCount, &error));
-    if (b->rules == nullptr)
+    rules_ = static_cast<struct IPv4Range*>(tr_sys_file_map_for_reading(fd, 0, byteCount, &error));
+    if (rules_ == nullptr)
     {
         tr_logAddWarn(fmt::format(
             _("Couldn't read '{path}': {error} ({error_code})"),
-            fmt::arg("path", b->filename),
+            fmt::arg("path", getFilename()),
             fmt::arg("error", error->message),
             fmt::arg("error_code", error->code)));
         tr_sys_file_close(fd);
         tr_error_free(error);
         return;
     }
 
-    b->fd = fd;
-    b->byteCount = byteCount;
-    b->ruleCount = byteCount / sizeof(struct tr_ipv4_range);
+    fd_ = fd;
+    byte_count_ = byteCount;
+    rule_count_ = byteCount / sizeof(IPv4Range);
 
     tr_logAddInfo(fmt::format(
-        ngettext("Blocklist '{path}' has {count} entry", "Blocklist '{path}' has {count} entries", b->ruleCount),
-        fmt::arg("path", tr_sys_path_basename(b->filename)),
-        fmt::arg("count", b->ruleCount)));
+        ngettext("Blocklist '{path}' has {count} entry", "Blocklist '{path}' has {count} entries", rule_count_),
+        fmt::arg("path", tr_sys_path_basename(getFilename())),
+        fmt::arg("count", rule_count_)));
 }
 
-static void blocklistEnsureLoaded(tr_blocklistFile* b)
+void BlocklistFile::ensureLoaded()
 {
-    if (b->rules == nullptr)
+    if (rules_ == nullptr)
     {
-        blocklistLoad(b);
+        load();
     }
 }
 
-static int compareAddressToRange(void const* va, void const* vb)
-{
-    auto const* a = static_cast<uint32_t const*>(va);
-    auto const* b = static_cast<struct tr_ipv4_range const*>(vb);
-
-    if (*a < b->begin)
-    {
-        return -1;
-    }
-
-    if (*a > b->end)
-    {
-        return 1;
-    }
-
-    return 0;
-}
-
-static void blocklistDelete(tr_blocklistFile* b)
-{
-    blocklistClose(b);
-    tr_sys_path_remove(b->filename);
-}
+// TODO: unused
+//static void blocklistDelete(tr_blocklistFile* b)
+//{
+//    blocklistClose(b);
+//    tr_sys_path_remove(b->filename, nullptr);
+//}
 
 /***
 ****  PACKAGE-VISIBLE
 ***/
 
-tr_blocklistFile* tr_blocklistFileNew(char const* filename, bool isEnabled)
+char const* BlocklistFile::getFilename() const
```

Comment:
this is more of an opinion thing than one approach being right or wrong, but I'd be fine with moving these short functions into the header -- e.g. ones that are just a couple of lines and have no if/else branches, e.g. getFilename(), ~BlocklistFile(), exists(), getRuleCount(), setEnabled()

Reply:
In the header the compiler has some more freedom to inline and optimize together with the calling code. In module body it probably won't have that. I'm happy to move small short stuff up to the header.